pkgs/development/compilers/ghc/common-hadrian.nix

```diff
@@ -426,9 +427,11 @@ stdenv.mkDerivation ({
   # GHC currently ships an edited config.sub so ghcjs is accepted which we can not rollback
   ${if targetPlatform.isGhcjs then "dontUpdateAutotoolsGnuConfigScripts" else null} = true;
 
+  # FreeBSD prebuilts are built for x86_64-portbld-freebsd instead of x86_64-unknown-freebsd
```

Comment:
arch-portbld-freebsd is the triple they use for freebsd ports builds

Reply:
I guess this should be a simple of case "choice of vendor should never matter".

Reply:
FWIW https://gitlab.haskell.org/ghc/ghc/-/issues/15718

Reply:
On master this should do it
```diff
diff --git a/configure.ac b/configure.ac
index 70e2d5a196..980c578ac4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -767,18 +767,6 @@ AC_ARG_ENABLE(dtrace,

 HaveDtrace=NO

-AC_PATH_PROG(DtraceCmd,dtrace)
-if test "x$EnableDtrace" = "xyes"; then
-  if test -n "$DtraceCmd"; then
-    if test "x$TargetOS_CPP-$TargetVendor_CPP" = "xdarwin-apple" \
-      -o "x$TargetOS_CPP-$TargetVendor_CPP" = "xfreebsd-portbld" \
-      -o "x$TargetOS_CPP-$TargetVendor_CPP" = "xsolaris2-unknown"; then
-      HaveDtrace=YES
-    fi
-  fi
-fi
-AC_SUBST(HaveDtrace)
-
 AC_PATH_PROG(HSCOLOUR,HsColour)
 # HsColour is passed to Cabal, so we need a native path
 if test "$HostOS"      = "mingw32" && \
diff --git a/m4/fptools_set_c_ld_flags.m4 b/m4/fptools_set_c_ld_flags.m4
index 13913709c9..8d51d350cc 100644
--- a/m4/fptools_set_c_ld_flags.m4
+++ b/m4/fptools_set_c_ld_flags.m4
@@ -33,7 +33,7 @@ AC_DEFUN([FPTOOLS_SET_C_LD_FLAGS],
     esac

     case $$1 in
-    i386-portbld-freebsd*)
+    i386-*-freebsd*)
         $2="$$2 -march=i686"
         ;;
     x86_64-unknown-solaris2)
```

`HaveDtrace` is dead code since the make build system was deleted leaving just Hadrian.